testdata/config/stackdriver.yaml

```diff
@@ -55,34 +54,116 @@ spec:
   actions:
   - handler: stackdriver.stackdriver.istio-config-default
     instances:
-    - requestcount.metric.istio-config-default
-    - requestduration.metric.istio-config-default
-    - global.logentry.stackdriver.istio-config-default
-
+    - stackdriver-requestcount.metric.istio-config-default
+    - stackdriver-requestduration.metric.istio-config-default
+    - stackdriver-requestsize.metric.istio-config-default
+    - stackdriver-responsesize.metric.istio-config-default
+    - stackdriver-global-mr.logentry.istio-config-default
 ---
 apiVersion: "config.istio.io/v1alpha2"
 kind: logentry
 metadata:
-  name: global
-  namespace: stackdriver.istio-config-default
+  name: stackdriver-global-mr
+  namespace: istio-config-default
 spec:
+  severity: '"Default"'
   timestamp: request.time
   variables:
-    originIp: origin.ip
-    targetIp: target.ip
-    sourceUser: origin.user
-    method: request.method
-    url: request.path
-    protocol: request.scheme
-    responseCode: response.code
-    responseSize: response.size
-    requestSize: request.size
-    latency: response.duration
+    originIp: origin.ip | ip("0.0.0.0")
+    targetIp: target.ip | ip("0.0.0.0")
+    sourceUser: origin.user | ""
+    method: request.method | ""
+    url: request.path | ""
+    protocol: request.scheme | "http"
+    responseCode: response.code | 0
+    responseSize: response.size | 0
+    requestSize: request.size | 0
+    latency: response.duration | "0ms"
   # The Monitored Resource must match a Stackdriver Monitored resource type defined at:
   #     https://cloud.google.com/monitoring/api/resources
   # Sending invalid MRs will result in the entire metric entry being rejected.
-  monitoredResourceType: "global"
+  monitoredResourceType: '"global"'
+  monitoredResourceDimensions:
+    # When using the global MR, this must be filled in, preferably matching the
+    # project_id given above in stackdriver config.
+    project_id: '"<your project ID>"'
+---
+apiVersion: "config.istio.io/v1alpha2"
+kind: metric
+metadata:
+  name: stackdriver-requestcount
```

Comment:
do we need / want a comment somewhere in our jumble of configs to indicate that we are trying to keep these definitions in sync with the non `stackdriver-` metrics?

also, i think we are avoiding `-`s entirely in names now.  @mandarjog is more knowledgeable on the details there, however.

Reply:
Added comment. Pinged mjog about naming.